packages/foam-core/src/utils.ts

```diff
@@ -1,4 +1,39 @@
+import path from 'path';
+import crypto from 'crypto';
 import { titleCase } from 'title-case';
+import GithubSlugger from 'github-slugger';
+import { URI, ID } from 'types';
+
+export const hash = (text: string) =>
+  crypto
+    .createHash('md5')
+    .update(text)
+    .digest('hex');
+
+export const uriToSlug = (noteUri: URI): string => {
+  // TODO hack don't merge with this
+  const filename = noteUri
+    .split('/')
+    .slice(-1)[0]
+    .split('.')
+    .slice(0, -1)
+    .join('.');
+  return GithubSlugger.slug(filename);
+};
+
+export const hashURI = (uri: URI): ID => {
+  return hash(path.normalize(uri));
+};
+
+export const getUriViaRelative = (
```

Comment:
This method probably needs a better name, since on a first read I'm struggling to understand what it does. Will probably become clear on second read.

Reply:
haha I totally agree :) feel free to suggest better names, will come up with something tomorrow